src/humanize/number.py

```diff
@@ -26,6 +26,17 @@
 NumberOrString: TypeAlias = "float | str"
 
 
+def format_non_finite(value: float) -> str:
+    """Utility function to handle infinite and nan cases."""
+    if math.isnan(value):
+        return "NaN"
+    if math.isinf(value) and value < 0:
+        return "-Inf"
+    elif math.isinf(value) and value > 0:
+        return "+Inf"
+    return ""
```

Comment:
This function seems overly complicated and not necessary to me. Why not just use the built-in string conversions? "nan", "-inf", and "inf"?

```python
if not math.isfinite(float(value)):
   return str(float(value))
```

Reply:
My argument for `NaN` was the capitalization in https://en.wikipedia.org/wiki/NaN, which I also prefer. But I don't have a strong opinion on that.

Reply:
Then what was the whole point of all this?...You could have just told me to do that at first...I'm giving up if this is what you don't want!

Reply:
Then what should I do please say!!
I'm changing the same thing over and over again!